aquadoggo/src/graphql/schema_builders/document.rs

```diff
@@ -0,0 +1,474 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use async_graphql::dynamic::{Field, FieldFuture, InputValue, Object, TypeRef};
+use dynamic_graphql::{Error, FieldValue, ScalarValue};
+use log::debug;
+use p2panda_rs::storage_provider::traits::DocumentStore;
+use p2panda_rs::{document::traits::AsDocument, schema::Schema};
+
+use crate::db::SqlStore;
+use crate::graphql::scalars::{DocumentIdScalar, DocumentViewIdScalar};
+use crate::graphql::types::DocumentMeta;
+use crate::graphql::utils::{downcast_id_params, fields_name, get_document_from_params};
+
+const QUERY_ALL_PREFIX: &str = "all_";
+const FIELDS_FIELD: &str = "fields";
+const META_FIELD: &str = "meta";
+const DOCUMENT_META_SCHEMA_ID: &str = "DocumentMeta";
+const DOCUMENT_ID_ARG: &str = "id";
+const DOCUMENT_VIEW_ID_ARG: &str = "viewId";
+const DOCUMENT_ID_SCALAR: &str = "DocumentId";
+const DOCUMENT_VIEW_ID_SCALAR: &str = "DocumentViewId";
```

Comment:
Maybe add some documentation here what these constants are representing? I assume that the ones ending on "_SCALAR" need to match what is defined in the regarding `Scalar` trait implementations?

Reply:
Great point, I'll add some docs.